metis-enrichment/metis-enrichment-client/src/test/java/eu/europeana/enrichment/rest/client/EnrichmentWorkerTest.java

```diff
@@ -0,0 +1,236 @@
+package eu.europeana.enrichment.rest.client;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.jibx.runtime.JiBXException;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.enrichment.api.external.model.Agent;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.EnrichmentResultList;
+import eu.europeana.enrichment.api.external.model.Place;
+import eu.europeana.enrichment.api.external.model.Timespan;
+import eu.europeana.enrichment.rest.client.EnrichmentWorker.Mode;
+import eu.europeana.enrichment.utils.EntityClass;
+import eu.europeana.enrichment.utils.EntityMergeEngine;
+import eu.europeana.enrichment.utils.InputValue;
+
+// TODO Tests that could be added:
+// 1. What if there is nothing to process (empty or null lists are returned when extracting or when
+// enriching/dereferencing).
+// 2. Throwing exceptions
+public class EnrichmentWorkerTest {
+
+  private static final InputValue[] ENRICHMENT_EXTRACT_RESULT =
+      new InputValue[] {new InputValue("orig1", "value1", "lang1", EntityClass.AGENT),
+          new InputValue(null, "value2", null, EntityClass.AGENT, EntityClass.CONCEPT),
+          new InputValue("orig3", null, "lang2")};
+
+  private static final String[] DEREFERENCE_EXTRACT_RESULT =
+      new String[] {"enrich1", "enrich3", null, "enrich4"};
+
+  @Test
+  public void testEnrichmentWorkerHappyFlow() throws DereferenceOrEnrichException {
+    for (Mode mode : Mode.values()) {
+      testEnrichmentWorkerHappyFlow(mode);
+    }
+  }
+
+  private void testEnrichmentWorkerHappyFlow(Mode mode) throws DereferenceOrEnrichException {
```

Comment:
Can this be split down to more specialized unit tests?

Reply:
Not so easy without restrucuring the EnrichmentWorker class. But I did split this test method up into smaller methods, hopefully that makes it a bit more clear.